### ISSUE TITLE
KeyError for [PrivateDNSName] - Property does not exist for currently-launching instances.

### ISSUE THREAD

**Issue by hirocrafts** [lyft/cartography#1161]
*Title*: Key Error

*Description*:
https://boto3.amazonaws.com/v1/documentation/api/1.9.42/reference/services/ec2.html#EC2.Client.describe_instances
PrivateDnsName (string) --
(IPv4 only) The private DNS hostname name assigned to the instance. This DNS hostname can only be used inside the Amazon EC2 network. This name is not available until the instance enters the running state.

[optional *Relevant Links*:]
> (https://boto3.amazonaws.com/v1/documentation/api/1.9.42/reference/services/ec2.html#EC2.Client.describe_instances)

*Logs*:
 ```WARNING:cartography.intel.aws:Caught exception syncing account [<redacted>]. aws-best-effort-mode is on so we are continuing on to the next AWS account. All exceptions will be aggregated and re-logged at the end of the sync.
Traceback (most recent call last):
  File "[<redacted>]cartography/intel/aws/__init__.py", line 169, in _sync_multiple_accounts
    _sync_one_account(
  File "[<redacted>]cartography/intel/aws/__init__.py", line 64, in _sync_one_account
    RESOURCE_FUNCTIONS[func_name](**sync_args)
  File "[<redacted>]cartography/util.py", line 165, in timed
    return method(*args, **kwargs)
  File "[<redacted>]cartography/intel/aws/ec2/instances.py", line 325, in sync_ec2_instances
    ec2_data = transform_ec2_instances(reservations, region, current_aws_account_id)
  File "[<redacted>]cartography/intel/aws/ec2/instances.py", line 130, in transform_ec2_instances
    'PrivateDnsName': network_interface['PrivateDnsName'],
KeyError: 'PrivateDnsName'
ERROR:cartography.intel.aws:AWS sync failed for accounts ['[<redacted>]']
ERROR:cartography.sync:Unhandled exception during sync stage 'aws'
Traceback (most recent call last):
  File "[<redacted>]cartography/sync.py", line 103, in run
    stage_func(neo4j_session, config)
  File "[<redacted>]cartography/util.py", line 165, in timed
    return method(*args, **kwargs)
  File "[<redacted>]cartography/intel/aws/__init__.py", line 298, in start_aws_ingestion
    sync_successful = _sync_multiple_accounts(
  File "[<redacted>]cartography/intel/aws/__init__.py", line 196, in _sync_multiple_accounts
    raise Exception('\n'.join(exception_tracebacks))
Exception: 2023-04-21 09:16:55.348526 - Exception for account ID: [<redacted>]
Traceback (most recent call last):
  File "[<redacted>]cartography/intel/aws/__init__.py", line 169, in _sync_multiple_accounts
    _sync_one_account(
  File "[<redacted>]cartography/intel/aws/__init__.py", line 64, in _sync_one_account
    RESOURCE_FUNCTIONS[func_name](**sync_args)
  File "[<redacted>]cartography/util.py", line 165, in timed
    return method(*args, **kwargs)
  File "[<redacted>]cartography/intel/aws/ec2/instances.py", line 325, in sync_ec2_instances
    ec2_data = transform_ec2_instances(reservations, region, current_aws_account_id)
  File "[<redacted>]cartography/intel/aws/ec2/instances.py", line 130, in transform_ec2_instances
    'PrivateDnsName': network_interface['PrivateDnsName'],
KeyError: 'PrivateDnsName'

Traceback (most recent call last):
  File "/Users/HVJD68/.virtualenvs/cartography-dev0/bin/cartography", line 33, in <module>
    sys.exit(load_entry_point('cartography', 'console_scripts', 'cartography')())
  File "[<redacted>]cartography/cli.py", line 600, in main
    sys.exit(CLI(prog='cartography').main(argv))
  File "[<redacted>]cartography/cli.py", line 581, in main
    return cartography.sync.run_with_config(self.sync, config)
  File "[<redacted>]cartography/sync.py", line 181, in run_with_config
    return sync.run(neo4j_driver, config)
  File "[<redacted>]cartography/sync.py", line 103, in run
    stage_func(neo4j_session, config)
  File "[<redacted>]cartography/util.py", line 165, in timed
    return method(*args, **kwargs)
  File "[<redacted>]cartography/intel/aws/__init__.py", line 298, in start_aws_ingestion
    sync_successful = _sync_multiple_accounts(
  File "[<redacted>]cartography/intel/aws/__init__.py", line 196, in _sync_multiple_accounts
    raise Exception('\n'.join(exception_tracebacks))
Exception: 2023-04-21 09:16:55.348526 - Exception for account ID: [<redacted>]
Traceback (most recent call last):
  File "[<redacted>]cartography/intel/aws/__init__.py", line 169, in _sync_multiple_accounts
    _sync_one_account(
  File "[<redacted>]cartography/intel/aws/__init__.py", line 64, in _sync_one_account
    RESOURCE_FUNCTIONS[func_name](**sync_args)
  File "[<redacted>]cartography/util.py", line 165, in timed
    return method(*args, **kwargs)
  File "[<redacted>]cartography/intel/aws/ec2/instances.py", line 325, in sync_ec2_instances
    ec2_data = transform_ec2_instances(reservations, region, current_aws_account_id)
  File "[<redacted>]cartography/intel/aws/ec2/instances.py", line 130, in transform_ec2_instances
    'PrivateDnsName': network_interface['PrivateDnsName'],
KeyError: 'PrivateDnsName'
```

*Please complete the following information:*:
>  - Cartography Version: 0.77.0
> - Commit Hash: 301601ca3c2be0e5dbde80b87cd73c4f9fc5b3ca
>  - Python version: 3.10.11
>  - OS: OSX Ventura 13.3.1

*Additional context*:
> Performed during multi-account sync. -- mentioned issue in Cartography slack.

**Comment by ramonpetgrave64** [comment]
Thanks for opening. There's an attempt to fix in #1162 , but can you verify that the particular instance was launched but not yet running at the time of this error?

**Comment by ramonpetgrave64** [comment]
Slack thread, for context
https://lyftoss.slack.com/archives/CTZUQL0KX/p1682087023164929
